### PR TITLE
Additions, rewording, clarifications

### PR DESCRIPTION
**Pull request by encukou**
To get back into writing the portingguide, I went through it and made some edits to hopefully make things clearer., more consistent, etc

### CODE REVIEW

**Comment by encukou** [comment]
I'm expanding the scope of this PR to avoid conflicts as I'm working on top of the fixes. See the individual commits.

File: source/strings.rst [hunk 1]
```diff
@@ -35,12 +35,12 @@ In Python 2, the ``str`` type was used for two different kinds of values –
 Ideally, every “stringy” value will explicitly and unambiguously be one of
 these types (or the native string, below).
 This means that you need to go through the entire codebase, and decide
-these two types.
+which value is what type these two types.
```
Comment (irushchyshyn) [inline source/strings.rst]: `these two types` should probably just be removed.

File: source/numbers.rst [hunk 2]
```diff
@@ -87,19 +95,18 @@ Removal of the ``long`` type
 * :ref:`Fixer <python-modernize>`: ``python-modernize -wnf lib2to3.fixes.fix_long``
 * Prevalence: Common
 
-The ``long`` builtin no longer exists.
```
Comment (hroncok) [inline source/numbers.rst]: That line just got lost? Was that intentional?

File: source/builtins.rst [hunk 3]
```diff
@@ -66,10 +66,44 @@ After running it, examine the output to determine if any :func:`eval`
...
+
+In Python 2, :func:`file` was the type of an open file. It was used in two
+ways:
+
+* To open files, i.e. as an alias for :func:`open`. The documentation mentions
+  that ``open`` is more appropriate for this case.
+* To check if an object is a file, as in ``isinstance(f, file)``.
+
+The recommended fixer addresses the first use: it will rewrite all calls to
+``file()`` to ``open()``. If
```
Comment (hroncok) [inline source/builtins.rst]: If what?

File: source/builtins.rst [hunk 4]
```diff
@@ -134,7 +134,20 @@ in some of your modules, revert the fixer's changes in that module.
 ``reduce()``
 ~~~~~~~~~~~~
 
-XXX
+* :ref:`Fixer <python-modernize>`: ``python-modernize -wnf libmodernize.fixes.fix_reduce``
+* Prevalence: Uncommon
+
+In Python 2, the function :func:`apply` was built in.
```
Comment (hroncok) [inline source/builtins.rst]: copy paste error. apply should be reduce

Comment (hroncok) [reply source/builtins.rst]: commited

**Comment by encukou** [comment]
Thanks for checking; here's a fix for the two missing sentences.

**Comment by hroncok** [comment]
Sometimes, the word Fixer is a link:

```rst
* :ref:`Fixer <python-modernize>`: *None*
```
And sometimes not:

```rst
* Fixer: None
```

Also note the (missing) emphasis on None.

Should I go and sync it to the first variant?

**Comment by encukou** [comment]
That would be great, if you have the time.

**Comment by hroncok** [comment]
I went with:

```rst
* :ref:`Fixer <python-modernize>`: None
```

(no emphasis, as I like it more without it)

**Comment by hroncok** [comment]
I've read the entire guide with this PR applied. I've fixed the stuff I found or opened an issue. So other than that I see no obvious problems.

**Comment by encukou** [comment]
Thanks! Let's do more fixes in PRs, then.

**Pull request merged**